anker/src/state.rs

```diff
@@ -566,6 +570,13 @@ impl Anker {
         )?;
         Ok(Box::new(wormhole_transfer_args))
     }
+
+    /// Get the `amount` of tokens from the SPL account defined by `account`.
+    /// Does not perform any checks, fails if not able to decode an SPL account.
```

Comment:
We could at least check that the account is owned by the SPL token program before we deserialize it.